src/main/scala/it/pagopa/interop/purposeprocess/common/readmodel/ReadModelQueries.scala

```diff
@@ -114,14 +118,16 @@ object ReadModelQueries {
     )(Filters.and).getOrElse(Filters.empty())
   }
 
-  private def listPurposesAuthorizationFilters(requesterId: UUID): Bson = {
-    // Include draft purposes only if the requester is the consumer
+  private def listPurposesAuthorizationFilters(excludeDraft: Option[Boolean]): Bson = {
+    // Exclude draft purposes only if requested
     // Note: the filter works on the assumption that if a version in Draft exists, it is the only version in the Purpose
-    val versionsFilter = Filters.or(
-      Filters
-        .and(Filters.eq("data.consumerId", requesterId.toString), Filters.eq("data.versions.state", Draft.toString)),
-      Filters.ne("data.versions.state", Draft.toString)
-    )
+    val versionsFilterWithoutDraft: Bson = Filters.and(Filters.ne("data.versions.state", Draft.toString))
+    val versionsFilterWithDraft: Bson    = Filters.and(Filters.eq("data.versions.state", Draft.toString))
```

Comment:
a request with `excludeDraft == false` should not force the presence of a Draft, but include all versions despite the state.
I would expect this filter to be
```suggestion
    val versionsFilterWithDraft: Bson    = Filters.empty
```